### ISSUE TITLE
Where to call call_MODIS

### ISSUE THREAD

**Issue by annethomas** [PecanProject/pecan#1611]
call_MODIS in data.remote is currently stand-alone and doesn't register files with the database. Where should this be done? Does convert_inputs have the necessary logic?

**Comment by annethomas** [comment]
Also, sidenote, it would be nice to have a progress bar on the LAI download in the python script even when debugging statements are turned off; it takes a long time for any decent number of pixels.

**Comment by annethomas** [comment]
@mdietze Is there a MODIS LAI format in the database? If not one will likely need to be made. Currently the python script underneath call_MODIS writes a netcdf, so if anything needs to change about the format that's where to look. For example, the dates in the output are the year and the day of year, like 2012001. If we want the date format to be more compatible with other functions, there are R functions that could easily change this, but I'm less familiar with what python can do.
(call_MODIS also currently expects that date format as input so it might be nice to take the usual date format in the arguments and reformat it within the function.)

**Comment by mdietze** [comment]
@annethomas good point, there may not be a MODIS LAI format yet. 👍 if you could create that.

In terms of date, it would be great if that could be in standard POSIX format, but if not you'll want to make sure you put the correct POSIX processing codes in the Format record and verify that they are loaded correctly by load_data.  Actually, I think you need to put in the POSIX codes regardless, so PEcAn know that's the datatime variable, but if the dates have a CF standard name and format then other tools (e.g. ncview, panoply) should be able to visualize as well

**Comment by annethomas** [comment]
@mdietze This is the way the netcdf is formatted in the python script, and I'm not sure it will work for load_data as is because it looks for a "time" dimension, and here it's "dates". I could just change the name of the dimension, but it's also only assigned to a variable that also holds dates, while LAI is just in terms of nrow and ncol (because of the spatial grid option). Would it make sense to get rid of the dates variable and move the time dimension to the LAI variables? How would that work with rows and columns? (I'm still wrapping my mind around multi-dimensional netcdfs.)
```
def m_data_to_netCDF(filename, m, k):
	rootgrp = netCDF4.Dataset(filename, 'w', format='NETCDF3_64BIT')
	rootgrp.createDimension('ncol', m.data.shape[1])
	rootgrp.createDimension('nrow', m.data.shape[0])
	rootgrp.createDimension('dates', len(m.dateInt))
	m_data = rootgrp.createVariable('LAI', 'f8', ('nrow', 'ncol'))
	m_std = rootgrp.createVariable('LAIStd', 'f8', ('nrow', 'ncol'))
	m_date = rootgrp.createVariable('Dates', 'i7', ('dates'))
	m_data[:] = m.data
	if k is not None:
        	m_std[:] = 0.1*k.data
	m_date[:] = m.dateInt
	rootgrp.close()
```
https://github.com/PecanProject/pecan/blob/develop/modules/data.remote/inst/modisWSDL.py#L340

**Comment by annethomas** [comment]
I've been looking at the values and I don't think ncol and nrow even represent the spatial grid; nrow is the number of dates and ncol is the linearized grid. So I'll experiment with giving it the number of km above and below instead along with time. Let me know if that doesn't make sense for some reason, and if the separate dates variable should also be kept (like the raw unformatted dates or something).

**Comment by mdietze** [comment]
Ultimately, I think it would be great to have a well formatted netCDF that has the LAI (and LAIsd) in a 3D array indexed by lat, lon, and time or by easting and northing and time. If you keep east and north relative to grid center, rather than lat/lon, it might be worth converting to UTM. As with UTM, I'd specify offsets in meters not km.

**Comment by annethomas** [comment]
Yeah, lat/lon would be great. I don't know that I have time to figure that out at the moment though, so is it ok if I do what I can with nrow/ncol and leave this issue open? I think my LAI benchmarks will all be single-pixel (linear over time), right?

**Comment by mdietze** [comment]
Yeah, if you're just doing single pixel then the format ends up essentially the same (or close enough that R will read it). I'm OK with leaving it open for now.

**Comment by annethomas** [comment]
@mdietze Do we want the time units to be days since the first year of the data or something else?

**Comment by mdietze** [comment]
I'll punt that to @bcow who knows more about how the date processing code in load_data is written, and how flexible it is for different data formatting, but it looks like the relevant code is here: https://github.com/PecanProject/pecan/blob/develop/modules/benchmark/R/load_netcdf.R#L35-L58
